### PR TITLE
Fix win64 debugger

### PR DESCRIPTION
**Pull request by skuater**
- radare2 for win64 now debug correctly a x32 program

### CODE REVIEW

File: libr/core/cmd_open.c [hunk 1]
```diff
@@ -236,8 +236,7 @@ static void cmd_open_map (RCore *core, const char *input) {
 	}
 	r_core_block_read (core);
 }
-
-R_API void r_core_file_reopen_in_malloc(RCore *core) {
+R_API void r_core_file_reopen_in_malloc (RCore *core) {
```
Comment (alvarofe) [inline libr/core/cmd_open.c]: I think this makes more sense in libr/core/file.c. I see in the first commit you did it, why did you regret about it?

Comment (skuater) [inline libr/core/cmd_open.c]: work fine on windows, but give compilations error on linux.

File: libr/core/cmd_open.c [hunk 2]
```diff
@@ -263,7 +263,7 @@ R_API void r_core_file_reopen_in_malloc(RCore *core) {
 	}
 }
 
-R_API void r_core_file_reopen_debug(RCore *core, const char *args) {
+R_API void r_core_file_reopen_debug (RCore *core, const char *args) {
```
Comment (radare) [inline libr/core/cmd_open.c]: wrong indent

File: libr/debug/p/native/maps/windows.c [hunk 3]
```diff
@@ -64,46 +65,33 @@ static RList *w32_dbg_maps(RDebug *dbg) {
 		CloseHandle (hModuleSnap);
 		return NULL;
 	}
-	hProcess=w32_openprocess (PROCESS_QUERY_INFORMATION |PROCESS_VM_READ,
-				FALSE, pid );
+	hProcess=w32_openprocess (PROCESS_QUERY_INFORMATION |PROCESS_VM_READ, FALSE, pid );
```
Comment (radare) [inline libr/debug/p/native/maps/windows.c]: wrong indent, spaces around = and |

File: libr/debug/p/native/maps/windows.c [hunk 4]
```diff
@@ -64,46 +65,33 @@ static RList *w32_dbg_maps(RDebug *dbg) {
...
+				nt_headers32 = (IMAGE_NT_HEADERS32 *)((char *)dos_header + dos_header->e_lfanew);
+				NumSections = nt_headers32->FileHeader.NumberOfSections;
+				SectionHeader = (IMAGE_SECTION_HEADER *)((char *)nt_headers32 + sizeof (IMAGE_NT_HEADERS32));
+
+			} else {
+				NumSections = nt_headers->FileHeader.NumberOfSections;
+				SectionHeader = (IMAGE_SECTION_HEADER *)((char *)nt_headers + sizeof (IMAGE_NT_HEADERS));
+			}
+			mr = r_debug_map_new (me32.szModule, (ut64)(size_t) (me32.modBaseAddr), (ut64)(size_t) (me32.modBaseAddr + SectionHeader->VirtualAddress), SectionHeader->Characteristics, 0);
 			if (mr != NULL) r_list_append (list, mr);
```
Comment (radare) [inline libr/debug/p/native/maps/windows.c]: remove != NULL and add braces

File: libr/debug/p/native/w32.c [hunk 5]
```diff
@@ -729,6 +730,9 @@ static int w32_dbg_wait(RDebug *dbg, int pid) {
 			break;
 		case EXCEPTION_DEBUG_EVENT:
 			switch (de.u.Exception.ExceptionRecord.ExceptionCode) {
+#if __MINGW64__
+			case 0x4000001f:
```
Comment (radare) [inline libr/debug/p/native/w32.c]: is this a missing EXCEPTION_ enum?

File: libr/core/cmd_open.c (hunk 1 above)
Comment (radare) [inline libr/core/cmd_open.c]: no it is not, it builds fine

Comment (skuater) [inline libr/core/cmd_open.c]: im talking about the first commit where some code was moved to file.c this is the reason of the second commit leaving this files untouched

File: libr/debug/p/native/w32.c (hunk 5 above)
Comment (skuater) [inline libr/debug/p/native/w32.c]: is a special exception for win32 subsystem, dont know if mingw have it defined in their sdk.

**Comment by radare** [comment]
Maybe use a const or define for this, do you know the name of it? Will merge when unlogic makes teavis happy again

**Comment by radare** [comment]
ping

File: libr/debug/p/native/maps/windows.c [hunk 6]
```diff
@@ -64,46 +65,33 @@ static RList *w32_dbg_maps(RDebug *dbg) {
...
-					SectionHeader->Name);
-
-				mr = r_debug_map_new (mapname,
-					(ut64)(size_t)(SectionHeader->VirtualAddress +\
-					me32.modBaseAddr),
-					(ut64)(size_t)(SectionHeader->VirtualAddress + \
-					me32.modBaseAddr + SectionHeader->Misc.VirtualSize),
-					SectionHeader->Characteristics, // XXX?
-					0);
+				sprintf(mapname,"%s | %s", me32.szModule, SectionHeader->Name);
```
Comment (radare) [inline libr/debug/p/native/maps/windows.c]: Missing space before (

File: libr/core/cmd_open.c [hunk 7]
```diff
@@ -310,6 +309,7 @@ R_API void r_core_file_reopen_debug(RCore *core, const char *args) {
 	free (newfile);
 }
 
+
```
Comment (radare) [inline libr/core/cmd_open.c]: Unnecessary empty line

File: libr/core/cconfig.c [hunk 8]
```diff
@@ -407,6 +407,13 @@ static int cb_asmbits(void *user, void *data) {
 		}
 		if (load_from_debug) {
 			if (core->dbg->h && core->dbg->h->reg_profile) {
+#if __WINDOWS__
+#if !defined(__MINGW64__)
```
Comment (radare) [inline libr/core/cconfig.c]: Better use positive conditions

**Pull request merged**